### ISSUE TITLE
No response when calling Inference Service

### ISSUE THREAD

**Issue by haksunLee** [kserve/kserve#1497]
/kind question

# Result

When called Inference service, it stops without response.

### When calling the service with Curl
```
# curl -v -H "Host: kfserving-demo-3-predictor-default.admin.example.com" -H "Cookie: authservice_session=MTYxNjU0NzUyNHxOd3dBTkVwVFQwMUlRelUyU1ZGQlZVMURRVUZTV0ZRMk5Gb3lSelJKVkVWYVNsTXpSMGxFVjBWUVIxVlVXVU5HUkVoWldGTlhWbEU9fL_pbxaK_BhAAx4_BHO5BLzqtFEO47v8nLzZBFqhATCz" http://10.0.2.15:32381/v1/models/kfserving-demo-3:predict -d @./input.json
* About to connect() to docker-image-registry port 32381 (#0)
*   Trying 10.0.2.15...
* Connected to docker-image-registry (10.0.2.15) port 32381 (#0)
> POST /v1/models/kfserving-demo-3:predict HTTP/1.1
> User-Agent: curl/7.29.0
> Accept: */*
> Host: kfserving-demo-3-predictor-default.admin.example.com
> Cookie: authservice_session=MTYxNjU0NzUyNHxOd3dBTkVwVFQwMUlRelUyU1ZGQlZVMURRVUZTV0ZRMk5Gb3lSelJKVkVWYVNsTXpSMGxFVjBWUVIxVlVXVU5HUkVoWldGTlhWbEU9fL_pbxaK_BhAAx4_BHO5BLzqtFEO47v8nLzZBFqhATCz
> Content-Length: 2523
> Content-Type: application/x-www-form-urlencoded
> Expect: 100-continue
> 
< HTTP/1.1 100 Continue
< HTTP/1.1 504 Gateway Timeout
< content-length: 24
< content-type: text/plain
< date: Thu, 25 Mar 2021 00:54:23 GMT
< server: istio-envoy
* HTTP error before end of send, stop sending
< 
* Closing connection 0

```

### When calling the Inference service through the Web UI
```
│ INFO|2021-03-24T11:10:07|/home/flask_server.py|38| Request args: {'model': u'kfserving-demo-3', 'name': u'kfserving-demo-3-predictor-default.admin.example.com', 'addr': u'10.0.2.15:32381'}                     
│ INFO|2021-03-24T11:10:07|/home/flask_server.py|43| img_path static/tmp/0b0903af-4c31-457a-b8c8-6bb04183d8cd.png                                                                                                  
│ INFO|2021-03-24T11:10:08|/home/mnist_client.py|59|  request_url http://10.0.2.15:32381/v1/models/kfserving-demo-3:predict                                                                                        
│ INFO|2021-03-24T11:10:08|/home/mnist_client.py|65| json_x                                                                                                                                                        
│ INFO|2021-03-24T11:10:08|/home/mnist_client.py|66| {"instances": [{"flatten_input": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  
│ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,   
│ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,   
│ 0, 0, 0, 0, 31, 140, 193, 44, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 38, 146, 240, 254, 254, 228, 48, 77, 46, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 43, 230, 254, 25  
│ 4, 254, 254, 254, 241, 254, 197, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 21, 130, 254, 254, 254, 239, 252, 254, 254, 254, 254, 237, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 154, 254, 254, 249,  
│  104, 71, 198, 254, 254, 254, 234, 57, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 173, 252, 252, 206, 51, 120, 215, 254, 254, 254, 254, 184, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 112, 254, 254, 215,  
│  87, 247, 254, 254, 254, 254, 254, 217, 31, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 5, 231, 254, 254, 254, 254, 254, 236, 128, 196, 254, 254, 119, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 197, 254,   
│ 254, 245, 238, 131, 17, 46, 247, 254, 199, 14, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 16, 92, 88, 40, 0, 0, 12, 173, 254, 242, 38, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0  
│ , 95, 254, 254, 155, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 18, 210, 254, 225, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 197, 254, 254, 99, 0, 0, 0  
│ , 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 34, 242, 254, 179, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 27, 223, 254, 225, 30, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0  
│ , 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 77, 254, 255, 127, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 54, 238, 254, 248, 53, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0  
│ , 0, 11, 183, 254, 254, 231, 41, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 113, 254, 254, 230, 48, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 110, 239, 126,   
│ 25, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  
│  0, 0, 0[]}]}                                                                                                                                                                                                    
│ INFO|2021-03-24T11:19:46|/home/mnist_client.py|71| response:                                                                                                                                                     
│ INFO|2021-03-24T11:19:46|/home/mnist_client.py|72| 504                                                                                                                                                           
│ INFO|2021-03-24T11:19:46|/home/mnist_client.py|73| upstream request timeout                                                                                                                                      
│ INFO|2021-03-24T11:19:46|/home/flask_server.py|65| Exception occured: No JSON object could be decoded                                                                                                            
│ INFO|2021-03-24T11:19:46|/usr/local/lib/python2.7/dist-packages/werkzeug/_internal.py|113| 127.0.0.1 - - [24/Mar/2021 11:19:46] "GET /?model=kfserving-demo-3&name=kfserving-demo-3-predictor-default.admin.exa  
│ mple.com&addr=10.0.2.15%3A32381 HTTP/1.1" 200 -    
```

# Object Status

#### single node

```
# k get node -o wide
NAME                    STATUS   ROLES    AGE   VERSION    INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                KERNEL-VERSION           CONTAINER-RUNTIME
docker-image-registry   Ready    master   14h   v1.18.16   10.0.2.15     <none>        CentOS Linux 7 (Core)   3.10.0-1160.el7.x86_64   docker://20.10.1
```

#### Istio 
```
# k get svc -n istio-system -o wide
NAME                       TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                                                                                                                                                                   AGE   SELECTOR
authservice                ClusterIP   10.99.210.196    <none>        8080/TCP                                                                                                                                                                                  13h   app.kubernetes.io/component=oidc-authservice,app.kubernetes.io/name=oidc-authservice,app=authservice
cluster-local-gateway      ClusterIP   10.100.229.112   <none>        80/TCP,443/TCP,31400/TCP,15011/TCP,8060/TCP,15029/TCP,15030/TCP,15031/TCP,15032/TCP                                                                                                       13h   app=cluster-local-gateway,istio=cluster-local-gateway,kustomize.component=cluster-local-gateway
istio-citadel              ClusterIP   10.109.85.97     <none>        8060/TCP,15014/TCP                                                                                                                                                                        13h   istio=citadel
istio-galley               ClusterIP   10.109.222.143   <none>        443/TCP,15014/TCP,9901/TCP                                                                                                                                                                13h   istio=galley
istio-ingressgateway       NodePort    10.107.239.113   <none>        15020:31037/TCP,80:31380/TCP,443:31390/TCP,31400:31400/TCP,15029:31843/TCP,15030:31275/TCP,15031:31080/TCP,15032:30297/TCP,15443:31416/TCP                                                13h   app=istio-ingressgateway,istio=ingressgateway
istio-pilot                ClusterIP   10.110.22.6      <none>        15010/TCP,15011/TCP,8080/TCP,15014/TCP                                                                                                                                                    13h   istio=pilot
istio-policy               ClusterIP   10.101.92.210    <none>        9091/TCP,15004/TCP,15014/TCP                                                                                                                                                              13h   istio-mixer-type=policy,istio=mixer
istio-sidecar-injector     ClusterIP   10.111.224.45    <none>        443/TCP,15014/TCP                                                                                                                                                                         13h   istio=sidecar-injector
istio-telemetry            ClusterIP   10.103.69.109    <none>        9091/TCP,15004/TCP,15014/TCP,42422/TCP                                                                                                                                                    13h   istio-mixer-type=telemetry,istio=mixer
kfserving-ingressgateway   NodePort    10.98.1.163      <none>        15020:30397/TCP,80:32381/TCP,443:32391/TCP,31400:32400/TCP,15011:31423/TCP,8060:32210/TCP,853:32500/TCP,15029:32270/TCP,15030:32539/TCP,15031:32611/TCP,15032:30695/TCP,15443:31606/TCP   12h   app=kfserving-ingressgateway,kfserving=ingressgateway
prometheus                 ClusterIP   10.105.153.254   <none>        9090/TCP                                                                                                                                                                                  13h   app=prometheus
```

#### Object
```
# k get all -n admin 
NAME                                                                  READY   STATUS      RESTARTS   AGE
pod/kfserving-demo-3-predictor-default-xd6c8-deployment-895dfdlsthb   3/3     Running     65         12h

NAME                                                       TYPE           CLUSTER-IP       EXTERNAL-IP                                               PORT(S)                             AGE
service/kfserving-demo-3                                   ExternalName   <none>           cluster-local-gateway.istio-system.svc.cluster.local      <none>                              12h
service/kfserving-demo-3-predictor-default                 ExternalName   <none>           kfserving-ingressgateway.istio-system.svc.cluster.local   <none>                              12h
service/kfserving-demo-3-predictor-default-xd6c8           ClusterIP      10.101.49.81     <none>                                                    80/TCP                              12h
service/kfserving-demo-3-predictor-default-xd6c8-private   ClusterIP      10.103.77.198    <none>                                                    80/TCP,9090/TCP,9091/TCP,8022/TCP   12h

NAME                                                                  READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/kfserving-demo-3-predictor-default-xd6c8-deployment   1/1     1            1           12h

NAME                                                                             DESIRED   CURRENT   READY   AGE
replicaset.apps/kfserving-demo-3-predictor-default-xd6c8-deployment-895dfd9      1         1         1       12h

NAME                    READY   AGE
statefulset.apps/nb01   1/1     12h

NAME                          COMPLETIONS   DURATION   AGE
job.batch/fairing-job-t42bw   0/1           12h        12h

NAME                                                                    CONFIG NAME                          K8S SERVICE NAME                           GENERATION   READY   REASON
revision.serving.knative.dev/kfserving-demo-3-predictor-default-xd6c8   kfserving-demo-3-predictor-default   kfserving-demo-3-predictor-default-xd6c8   1            True    

NAME                                                           URL                                                           READY   REASON
route.serving.knative.dev/kfserving-demo-3-predictor-default   http://kfserving-demo-3-predictor-default.admin.example.com   True    

NAME                                                                   LATESTCREATED                              LATESTREADY                                READY   REASON
configuration.serving.knative.dev/kfserving-demo-3-predictor-default   kfserving-demo-3-predictor-default-xd6c8   kfserving-demo-3-predictor-default-xd6c8   True    

NAME                                                             URL                                                           LATESTCREATED                              LATESTREADY                                READY   REASON
service.serving.knative.dev/kfserving-demo-3-predictor-default   http://kfserving-demo-3-predictor-default.admin.example.com   kfserving-demo-3-predictor-default-xd6c8   kfserving-demo-3-predictor-default-xd6c8   True    
```

#### knative inferenceservice 
```
# k get ksvc -n admin
NAME                                 URL                                                           LATESTCREATED                              LATESTREADY                                READY   REASON
kfserving-demo-3-predictor-default   http://kfserving-demo-3-predictor-default.admin.example.com   kfserving-demo-3-predictor-default-xd6c8   kfserving-demo-3-predictor-default-xd6c8   True    

# k get inferenceservice -n admin
NAME               URL                                         READY   DEFAULT TRAFFIC   CANARY TRAFFIC   AGE
kfserving-demo-3   http://kfserving-demo-3.admin.example.com   True    100                                12h

# k get revision -n admin
NAME                                       CONFIG NAME                          K8S SERVICE NAME                           GENERATION   READY   REASON
kfserving-demo-3-predictor-default-xd6c8   kfserving-demo-3-predictor-default   kfserving-demo-3-predictor-default-xd6c8   1            True    
```

#### Custom gateway 
```
# k get gateway -n knative-serving knative-ingress-gateway -o yaml
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"networking.istio.io/v1alpha3","kind":"Gateway","metadata":{"annotations":{},"labels":{"app.kubernetes.io/component":"knative-serving-install","app.kubernetes.io/name":"knative-serving-install","kustomize.component":"knative","networking.knative.dev/ingress-provider":"istio","serving.knative.dev/release":"v0.14.3"},"name":"knative-ingress-gateway","namespace":"knative-serving"},"spec":{"selector":{"app":"kfserving-ingressgateway","kfserving":"ingressgateway"},"servers":[{"hosts":["*"],"port":{"name":"http","number":80,"protocol":"HTTP"}}]}}
  creationTimestamp: "2021-03-24T10:28:28Z"
  generation: 1
  labels:
    app.kubernetes.io/component: knative-serving-install
    app.kubernetes.io/name: knative-serving-install
    kustomize.component: knative
    networking.knative.dev/ingress-provider: istio
    serving.knative.dev/release: v0.14.3
  managedFields:
  - apiVersion: networking.istio.io/v1alpha3
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:kubectl.kubernetes.io/last-applied-configuration: {}
        f:labels:
          .: {}
          f:app.kubernetes.io/component: {}
          f:app.kubernetes.io/name: {}
          f:kustomize.component: {}
          f:networking.knative.dev/ingress-provider: {}
          f:serving.knative.dev/release: {}
      f:spec:
        .: {}
        f:selector:
          .: {}
          f:app: {}
          f:kfserving: {}
        f:servers: {}
    manager: kubectl
    operation: Update
    time: "2021-03-24T10:28:28Z"
  name: knative-ingress-gateway
  namespace: knative-serving
  resourceVersion: "18209"
  selfLink: /apis/networking.istio.io/v1alpha3/namespaces/knative-serving/gateways/knative-ingress-gateway
  uid: 5f7792b7-392d-4b03-83fa-a0208ea85466
spec:
  selector:
    app: kfserving-ingressgateway
    kfserving: ingressgateway
  servers:
  - hosts:
    - '*'
    port:
      name: http
      number: 80
      protocol: HTTP


# k get virtualservice -A
NAMESPACE   NAME                                         GATEWAYS                                                                          HOSTS                                                                                                                                                                                                     AGE
admin       kfserving-demo-3                             [knative-ingress-gateway.knative-serving knative-serving/cluster-local-gateway]   [kfserving-demo-3.admin.example.com kfserving-demo-3.admin.svc.cluster.local]                                                                                                                             12h
admin       kfserving-demo-3-predictor-default-ingress   [knative-serving/cluster-local-gateway knative-serving/knative-ingress-gateway]   [kfserving-demo-3-predictor-default.admin kfserving-demo-3-predictor-default.admin.example.com kfserving-demo-3-predictor-default.admin.svc kfserving-demo-3-predictor-default.admin.svc.cluster.local]   12h
admin       kfserving-demo-3-predictor-default-mesh      [mesh]                                                                            [kfserving-demo-3-predictor-default.admin kfserving-demo-3-predictor-default.admin.svc kfserving-demo-3-predictor-default.admin.svc.cluster.local]                                                        12h


# k get virtualservice -n admin kfserving-demo-3-predictor-default-ingress -o yaml
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    networking.knative.dev/ingress.class: istio.ingress.networking.knative.dev
    serving.knative.dev/creator: system:serviceaccount:kubeflow:default
    serving.knative.dev/lastModifier: system:serviceaccount:kubeflow:default
  creationTimestamp: "2021-03-24T10:47:19Z"
  generation: 1
  labels:
    networking.internal.knative.dev/ingress: kfserving-demo-3-predictor-default
    serving.knative.dev/route: kfserving-demo-3-predictor-default
    serving.knative.dev/routeNamespace: admin
  managedFields:
  - apiVersion: networking.istio.io/v1alpha3
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:networking.knative.dev/ingress.class: {}
          f:serving.knative.dev/creator: {}
          f:serving.knative.dev/lastModifier: {}
        f:labels:
          .: {}
          f:networking.internal.knative.dev/ingress: {}
          f:serving.knative.dev/route: {}
          f:serving.knative.dev/routeNamespace: {}
        f:ownerReferences: {}
      f:spec:
        .: {}
        f:gateways: {}
        f:hosts: {}
        f:http: {}
    manager: controller
    operation: Update
    time: "2021-03-24T10:47:19Z"
  name: kfserving-demo-3-predictor-default-ingress
  namespace: admin
  ownerReferences:
  - apiVersion: networking.internal.knative.dev/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Ingress
    name: kfserving-demo-3-predictor-default
    uid: bd03166e-54e2-4db3-9431-9e39043387d4
  resourceVersion: "31577"
  selfLink: /apis/networking.istio.io/v1alpha3/namespaces/admin/virtualservices/kfserving-demo-3-predictor-default-ingress
  uid: 6cf04917-ef96-4e07-94b3-465250d66f43
spec:
  gateways:
  - knative-serving/cluster-local-gateway
  - knative-serving/knative-ingress-gateway
  hosts:
  - kfserving-demo-3-predictor-default.admin
  - kfserving-demo-3-predictor-default.admin.example.com
  - kfserving-demo-3-predictor-default.admin.svc
  - kfserving-demo-3-predictor-default.admin.svc.cluster.local
  http:
  - headers:
      request:
        set:
          K-Network-Hash: 2b3662633399698eeac14eedb96e93be831feb0d9c6a4f2e50195e9e46b3d178
    match:
    - authority:
        prefix: kfserving-demo-3-predictor-default.admin
      gateways:
      - knative-serving/cluster-local-gateway
    retries:
      attempts: 3
      perTryTimeout: 600s
      retryOn: 5xx,connect-failure,refused-stream,cancelled,resource-exhausted,retriable-status-codes
    route:
    - destination:
        host: kfserving-demo-3-predictor-default-xd6c8.admin.svc.cluster.local
        port:
          number: 80
      headers:
        request:
          set:
            Knative-Serving-Namespace: admin
            Knative-Serving-Revision: kfserving-demo-3-predictor-default-xd6c8
      weight: 100
    timeout: 600s
  - headers:
      request:
        set:
          K-Network-Hash: 2b3662633399698eeac14eedb96e93be831feb0d9c6a4f2e50195e9e46b3d178
    match:
    - authority:
        prefix: kfserving-demo-3-predictor-default.admin.example.com
      gateways:
      - knative-serving/knative-ingress-gateway
    retries:
      attempts: 3
      perTryTimeout: 600s
      retryOn: 5xx,connect-failure,refused-stream,cancelled,resource-exhausted,retriable-status-codes
    route:
    - destination:
        host: kfserving-demo-3-predictor-default-xd6c8.admin.svc.cluster.local
        port:
          number: 80
      headers:
        request:
          set:
            Knative-Serving-Namespace: admin
            Knative-Serving-Revision: kfserving-demo-3-predictor-default-xd6c8
      weight: 100
    timeout: 600s

# k get virtualservice -n admin kfserving-demo-3-predictor-default-mesh -o yaml
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    networking.knative.dev/ingress.class: istio.ingress.networking.knative.dev
    serving.knative.dev/creator: system:serviceaccount:kubeflow:default
    serving.knative.dev/lastModifier: system:serviceaccount:kubeflow:default
  creationTimestamp: "2021-03-24T10:47:19Z"
  generation: 1
  labels:
    networking.internal.knative.dev/ingress: kfserving-demo-3-predictor-default
    serving.knative.dev/route: kfserving-demo-3-predictor-default
    serving.knative.dev/routeNamespace: admin
  managedFields:
  - apiVersion: networking.istio.io/v1alpha3
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:networking.knative.dev/ingress.class: {}
          f:serving.knative.dev/creator: {}
          f:serving.knative.dev/lastModifier: {}
        f:labels:
          .: {}
          f:networking.internal.knative.dev/ingress: {}
          f:serving.knative.dev/route: {}
          f:serving.knative.dev/routeNamespace: {}
        f:ownerReferences: {}
      f:spec:
        .: {}
        f:gateways: {}
        f:hosts: {}
        f:http: {}
    manager: controller
    operation: Update
    time: "2021-03-24T10:47:19Z"
  name: kfserving-demo-3-predictor-default-mesh
  namespace: admin
  ownerReferences:
  - apiVersion: networking.internal.knative.dev/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Ingress
    name: kfserving-demo-3-predictor-default
    uid: bd03166e-54e2-4db3-9431-9e39043387d4
  resourceVersion: "31575"
  selfLink: /apis/networking.istio.io/v1alpha3/namespaces/admin/virtualservices/kfserving-demo-3-predictor-default-mesh
  uid: 00ee4d1b-3d2f-49fa-a0fe-118f7cdd9963
spec:
  gateways:
  - mesh
  hosts:
  - kfserving-demo-3-predictor-default.admin
  - kfserving-demo-3-predictor-default.admin.svc
  - kfserving-demo-3-predictor-default.admin.svc.cluster.local
  http:
  - headers:
      request:
        set:
          K-Network-Hash: 2b3662633399698eeac14eedb96e93be831feb0d9c6a4f2e50195e9e46b3d178
    match:
    - authority:
        prefix: kfserving-demo-3-predictor-default.admin
      gateways:
      - mesh
    retries:
      attempts: 3
      perTryTimeout: 600s
      retryOn: 5xx,connect-failure,refused-stream,cancelled,resource-exhausted,retriable-status-codes
    route:
    - destination:
        host: kfserving-demo-3-predictor-default-xd6c8.admin.svc.cluster.local
        port:
          number: 80
      headers:
        request:
          set:
            Knative-Serving-Namespace: admin
            Knative-Serving-Revision: kfserving-demo-3-predictor-default-xd6c8
      weight: 100
    timeout: 600s

```

#### Knative Config
```
# kubectl -n kubeflow edit cm inferenceservice-config

  ingress: |-
    {
        "ingressGateway" : "knative-ingress-gateway.knative-serving",
        "ingressService" : "kfserving-ingressgateway.istio-system.svc.cluster.local"
    }



# kubectl -n knative-serving edit cm config-istio

apiVersion: v1
data:
  gateway.knative-serving.knative-ingress-gateway: kfserving-ingressgateway.istio-system.svc.cluster.local
  local-gateway.knative-serving.cluster-local-gateway: kfserving-ingressgateway.istio-system.svc.cluster.local
  local-gateway.mesh: mesh
  reconcileExternalGateway: "false"
```

# Execution environment
```
Virtualbox 6.1
CentOs 7.9
Docker CE 20.10.1
K8s v1.18.16
kubeflow 1.2 (kfctl_istio_dex.v1.2.0)

gcr.io/knative-releases/knative.dev/serving/cmd/webhook        v0.14.3
gcr.io/knative-releases/knative.dev/serving/cmd/queue          <none> 
gcr.io/knative-releases/knative.dev/serving/cmd/activator      v0.14.3
gcr.io/knative-releases/knative.dev/serving/cmd/controller     v0.14.3
gcr.io/knative-releases/knative.dev/serving/cmd/autoscaler     v0.14.3
gcr.io/knative-releases/knative.dev/net-istio/cmd/webhook      v0.14.1
gcr.io/knative-releases/knative.dev/net-istio/cmd/controller   v0.14.1

gcr.io/kfserving/kfserving-controller                          v0.4.1
gcr.io/kfserving/storage-initializer                           v0.4.1

gcr.io/istio-release/proxyv2                                   release-1.3-latest-daily
gcr.io/istio-release/proxy_init                                release-1.3-latest-daily
gcr.io/istio-release/pilot                                     release-1.3-latest-daily
gcr.io/istio-release/node-agent-k8s                            release-1.3-latest-daily
gcr.io/istio-release/mixer                                     release-1.3-latest-daily
gcr.io/istio-release/kubectl                                   release-1.3-latest-daily
gcr.io/istio-release/galley                                    release-1.3-latest-daily
gcr.io/istio-release/citadel                                   release-1.3-latest-daily

```

**Comment by haksunLee** [comment]
sss

**Comment by haksunLee** [comment]
Inference Server is called through a custom gateway, but there is no response. Am I wrong setting or wrong calling method?

**Comment by haksunLee** [comment]
When called with the option sidecar.istio.io/inject: "false" in inferenceservice annotations, it worked normally. However, this is not the normal way. Do you know another solution?

**Comment by markwinter** [comment]
I'm having this same problem with a Kubeflow 1.2 setup.

I created a separate Istio Ingress Deployment, Service, Gateway.

Updated KFServing with https://github.com/kubeflow/kfserving/blob/master/install/v0.5.1/kfserving_kubeflow.yaml and edited the ingress options to point to my new Service+Gateway.

And I edited knative `config-istio`, kubeflow `kfserving-config` `inferenceservice-config` configmaps to use the new Ingress.

However the request just hangs when calling.

Is there some configuration I'm missing?
